### ISSUE TITLE
Extremely large SPDX files times out

### ISSUE THREAD

**Issue by goneall** [spdx/spdx-online-tools#306]
Suggest a short term solution where we have a size limit on SPDX documents for validate, translate, and convert then produce an error and suggest they use the command line tool available at github.com/spdx/tools-java

Longer term, we could do some kind of batch processing approach and not suffer the processing timeouts.

**Comment by goneall** [comment]
@dkruszew reported the same issue and included the attached file as a reference.

In looking at the file, it is quite large at 122,244 lines.

As a work-around, the latest commandline version of the SPDX online tools from the [release page](https://github.com/spdx/tools-java/releases/tag/v1.0.1) and run locally.

I tried this and it took 1 minute 46 seconds on a laptop.  This is processing 1,153 lines per second - slower that I would of expected but not crazy unreasonable.

I noticed you have a lot of relationships in this document.  One optimization that would make a potentially big difference is to cache the verification results of each element so we don't end up re-traversing the same nodes when we reference the same file.

I've added an issue to track this enhancment: https://github.com/spdx/tools-java/issues/20
[keycloak-13.0.1.zip-too-big.txt](https://github.com/spdx/spdx-online-tools/files/6712837/keycloak-13.0.1.zip-too-big.txt)
